python/ray/serve/pipeline/tests/test_json_serde.py

```diff
@@ -322,6 +328,44 @@ def test_nested_deployment_node_json_serde(serve_instance):
     )
 
 
+def get_handle(sync: bool = True):
+    @serve.deployment
+    def echo(inp: str):
+        return inp
+
+    echo.deploy()
+    return echo.get_handle(sync=sync)
+
+
+async def call(handle, inp):
+    if isinstance(handle, RayServeSyncHandle):
+        ref = handle.remote(inp)
+    else:
+        ref = await handle.remote(inp)
+
+    return ray.get(ref)
+
+
+@pytest.mark.asyncio
+class TestHandleJSON:
+    def test_invalid(self, serve_instance):
+        with pytest.raises(ValueError):
+            serve_handle_from_json_dict({"blah": 123})
+
+    @pytest.mark.parametrize("sync", [False, True])
+    async def test_basic(self, serve_instance, sync):
+        handle = get_handle(sync)
+        await call(handle, "hi") == "hi"
+
+        serialized = json.dumps(serve_handle_to_json_dict(handle))
```

Comment:
can we also add to ensure `json.dumps(handle, cls=DAGNodeEncoder)` works the same ?

Reply:
I already have some tests for this in the `serve build` PR, will leave them out here for simplicity